src/app/features/worklog/worklog.component.ts

```diff
@@ -34,8 +35,33 @@ export class WorklogComponent {
     private readonly _taskService: TaskService,
     private readonly _matDialog: MatDialog,
     private readonly _router: Router,
+    private readonly _route: ActivatedRoute,
   ) {}
 
+  ngAfterViewInit(): void {
+    this._route.queryParams.subscribe((params) => {
+      const { dateStr, focusItem } = params as SearchQueryParams;
+      if (!!dateStr) {
+        this.expanded[dateStr] = true;
+        this.focusTask(focusItem);
+      }
+    });
+  }
+
+  focusTask(id: string) {
```

Comment:
Could you maybe add a comment on why we need setInterval here rather than just a simple setTimeout? I'd assume that it is needed because the element might not be present yet?

Reply:
Exactly, just adjusting the timeout period once didn't really work nicely. Sometimes the page itself took more than a second to load and I didn't want to set the timeout period too high. This way it tries every 300ms up to 5 times. I didn't really manage to find a perfect way that works every time.
In this sense, the approach I took for normal tasks is the cleanest one so far. For my tests it has worked really well with a static period of only 200ms. This way the timeout gets set when one of the last and smallest components is created: the task itself. 
I find it hard to figure out when the page is finally completely loaded (including animations and maybe even the full-page-spinner might mess with this working correctly). 
This is why I moved the blendInOutAnimation a level deeper into the add-task-bar/search-bar component so I could hook to onAnimationEvent().
If you have a better solution for this page-load/event-hook problem, I'd love to hear about it. As it might be useful once I move the navigation part (getElementById + focus) to a parent component.

Reply:
Thinking about it: This might have introduced some race condition, which leads to the unwanted focus behaviour. Simple fix: Save the intervalId and cancel potentially running intervals whenever the function is executed.

A slightly cleaner solution might couple this to worklogData$, but then again KISS as it is not straightforward neither. :)

Reply:
> Thinking about it: This might have introduced some race condition, which leads to the unwanted focus behaviour. 
> Simple fix: Save the intervalId and cancel potentially running intervals whenever the function is executed.

I don't think so as it happened with a simple setTimeout, too. But I think we'll get to a solution for this later on. 

> A slightly cleaner solution might couple this to worklogData$, but then again KISS as it is not straightforward neither. :)

I have no idea what you mean here. Couple what to worklogData$? And why? This might also be a subject for later as I'd like to propose another idea. :)

Reply:
This should be fixed now with css update and interval clearer.